syntax = "proto3";

package optur;

message SamplerConfig {
    oneof sampler {
        RandomSamplerConfig random = 1;
        TPESamplerConfig tpe = 2;
    }
}


message KDEConfig {
    bool consider_prior = 1;
    double prior_weight = 2;
    bool consider_magic_clip = 3;
    bool consider_endpoints = 4;
}


message RandomSamplerConfig {}

message TPESamplerConfig {
    KDEConfig kde = 1;
    int64 n_startup_trials = 2;
    int64 n_ei_candidates = 3;
    SamplerConfig fallback_sampler = 4;
}
